fred2/sexp_tree.cpp

```diff
@@ -7348,6 +7356,25 @@ sexp_list_item *sexp_tree::get_container_multidim_modifiers(int con_data_node) c
 	return head.next;
 }
 
+sexp_list_item* sexp_tree::check_for_dynamic_sexp_enum(int opf)
+{
+	sexp_list_item head;
+
+	int item = opf - First_available_list_id;
+
+	if (item <= (int)Dynamic_enums.size()) {
+
+		for (int i = 0; i < (int)Dynamic_enums[item].list.size(); i++) {
+			head.add_data(Dynamic_enums[item].list[i].c_str());
```

Comment:
Just make this a for each loop:
```suggestion
		for (const SCP_string& enum_item : Dynamic_enums[item].list) {
			head.add_data(enum_item.c_str());
```